catalog/app/containers/Bucket/Listing.tsx

```diff
@@ -0,0 +1,850 @@
+import cx from 'classnames'
+import * as R from 'ramda'
+import * as React from 'react'
+import AutosizeInput from 'react-input-autosize'
+import type { AutosizeInputProps } from 'react-input-autosize'
+import { Link } from 'react-router-dom'
+import * as M from '@material-ui/core'
+import { fade } from '@material-ui/core/styles'
+import * as DG from '@material-ui/data-grid'
+
+import { renderPageRange } from 'components/Pagination2'
+import { readableBytes } from 'utils/string'
+
+const EMPTY = <i>{'<EMPTY>'}</i>
+
+const TIP_DELAY = 1000
+
+export interface Item {
+  type: 'dir' | 'file'
+  name: string
+  to: string
+  size?: number
+  modified?: Date
+  archived?: boolean
+}
+
+function maxPartial<T extends R.Ord>(a: T | undefined, b: T | undefined) {
+  if (a) return b ? R.max(a, b) : a
+  return b
+}
+
+const computeStats = R.reduce(
+  (acc, i: Item) => ({
+    files: i.type === 'file' ? acc.files + 1 : acc.files,
+    dirs: i.type === 'dir' && i.name !== '..' ? acc.dirs + 1 : acc.dirs,
+    size: i.size ? acc.size + i.size : acc.size,
+    modified: maxPartial(i.modified, acc.modified),
+  }),
+  {
+    dirs: 0,
+    files: 0,
+    size: 0,
+    modified: undefined as Date | undefined,
+  },
+)
+
+type DataGridProps = Omit<DG.GridComponentProps, 'licenseStatus'>
+
+const DataGrid = React.memo(
+  React.forwardRef<HTMLDivElement, DataGridProps>(function DataGrid(inProps, ref) {
+    const props = DG.useThemeProps({ props: inProps, name: 'MuiDataGrid' })
+    return <DG.GridComponent ref={ref} {...props} licenseStatus="Valid" />
+  }),
+)
+
+interface WrappedAutosizeInputProps extends Omit<AutosizeInputProps, 'ref'> {
+  className?: string
+}
+
+function WrappedAutosizeInput({ className, ...props }: WrappedAutosizeInputProps) {
+  return <AutosizeInput inputClassName={className} {...props} />
+}
+
+const usePrefixFilterStyles = M.makeStyles((t) => ({
+  root: {
+    fontSize: 14,
+    height: 20,
+    lineHeight: 20,
+    padding: 0,
+  },
+  btn: {
+    fontSize: 11,
+    lineHeight: '22px',
+    marginLeft: t.spacing(0.75),
+    minWidth: 'auto',
+    paddingBottom: 0,
+    paddingTop: 2,
+  },
+  clearIcon: {
+    fontSize: '16px !important',
+    lineHeight: '15px',
+    marginLeft: -6,
+  },
+  searchIcon: {
+    fontSize: 20,
+    marginLeft: -2,
+    marginRight: t.spacing(0.5) - 2,
+  },
+}))
+
+interface PrefixFilterProps {
+  prefix?: string
+  setPrefix: (prefix: string) => void
+}
+
+export function PrefixFilter({ prefix = '', setPrefix }: PrefixFilterProps) {
+  const classes = usePrefixFilterStyles()
+
+  const inputRef = React.useRef<{ blur: () => void }>()
+  const [prefixValue, setPrefixValue] = React.useState(prefix)
+
+  const blur = React.useCallback(() => {
+    inputRef.current?.blur?.()
```

Comment:
Just checked the resulting code from `./build` directory, it already contains a lot of chaining. But, it seems that chaining operators are transpiled into

![Screenshot from 2021-03-17 15-16-09](https://user-images.githubusercontent.com/533229/111466037-d1ed3e00-8733-11eb-8bfa-8acd22b7f115.png)

Reply:
that's bc we're down-leveling to ES5